java/ql/src/semmle/code/java/dataflow/internal/TaintTrackingUtil.qll

```diff
@@ -377,34 +377,19 @@ private predicate taintPreservingQualifierToMethod(Method m) {
   or
   m = any(ProtobufMessageLite p).getAGetterMethod()
   or
-  m instanceof MapMethod and
-  (
-    m.getName().regexpMatch("get|entrySet|keySet|values")
```

Comment:
Just noting that `keySet` is not covered in `ContainerFlow` (unlike the others). However, I believe this is fine for Spring, because a `Map` type parameter to a `@RequestMapping` method will be unlikely to contain tainted keys, and `ContainerFlow` only taints `Map`s via the `value` not the `key` parameter.

(ref: https://docs.spring.io/spring/docs/current/spring-framework-reference/web.html#mvc-ann-arguments)

Reply:
`keySet` is excluded on purpose as we currently otherwise won't be able to distinguish keys and values in a tainted map.  If at some point tainted keys become important to track, then we'll most likely attempt this through a more precise modelling via container-content-as-field-flow.